contracts/DistributionContract.sol

```diff
@@ -1,75 +1,184 @@
+    IERC20 private _token;
+
+    event RewardCollect(
+        address beneficiaries_,
+        uint256 amount_,
+        bool transferStatus_
+    );
+    event ClaimInfo(address beneficiaries_, uint256 amount_);
+    event LockStatus(bool lock);
+
+    /**
+     * @dev modifier which contains conditions who's can call functions
+     *
+     * NOTE : if function have 'onlyOwner' thats mean call this function
+     * can only address which contained in '_contractOwner'
+     */
+    modifier onlyOwner() {
+        require(_contractOwner == msg.sender, "caller is not the owner");
         _;
     }
 
-    function deposit (uint256 _amount) public onlyOwner {
-        _deposit[contractOwner] += _amount;
-        token.transferFrom(contractOwner, address(this), _amount);
+    /**
+     * @dev Set _token IERC20 to interact with thrid party token
+     *
+     * Put address to state variable '_contractOwner'
+     *
+     * @param myToken_ of ERC20 contract
+     */
+    constructor(address myToken_) {
+        _token = IERC20(myToken_);
+        _contractOwner = msg.sender;
     }
 
-    function addBeneficiaries (address _beneficiaries , uint256 _amount) public onlyOwner {
-
-        listOfBeneficiaries[_beneficiaries]._addressOfBeneficiary = _beneficiaries;
-        listOfBeneficiaries[_beneficiaries]._reward = _amount;
+    /**
+     * @dev send ERC20 tokens from '_contractOwner' balance to SC
+     *
+     * @param amount_ how many tokens send
+     *
+     * NOTE : this function use wrapper of safeERC20 library
+     * check 'SafeERC20.sol' for more information
+     */
+    function deposit(uint256 amount_) external onlyOwner {
+        _token.safeTransferFrom(msg.sender, address(this), amount_);
     }
 
-    function addBeneficiary (address _beneficiary , uint256 _amount) public onlyOwner {
-
-        listOfBeneficiaries[_beneficiary]._addressOfBeneficiary = _beneficiary;
-        listOfBeneficiaries[_beneficiary]._reward = _amount;
+    /**
+     * @dev filling beneficiaries balances
+     *
+     * @param beneficiaries_ , contains who's will get tokens
+     * @param amount_ , how many tokens will get
+     * Requirements:
+     *
+     * - `owner` cannot be the zero address.
+     * - `spender` cannot be the zero address.
+     * - 'beneficiaries_' & 'amount_' must have same length
+     */
+    function addBeneficiaries(
+        address[] calldata beneficiaries_,
+        uint256[] calldata amount_
+    ) external onlyOwner {
+        for (uint256 i; i < beneficiaries_.length; i++) {
+            require(
+                beneficiaries_[i] != address(0) && amount_[i] != 0,
+                "beneficiary or amount equal to 0"
+            );
+            require(
+                beneficiaries_.length == amount_.length,
+                "arrays have different length"
+            );
+            balanceOfBeneficiares[beneficiaries_[i]] += amount_[i];
+            emit RewardCollect(beneficiaries_[i], amount_[i], true);
```

Comment:
Also, go emit general event for method only one time after for. At the moment you emit for every iteration event